src/formats/bbcode.js

```diff
@@ -1800,7 +1800,7 @@
 		function convertToHTML(tokens, isRoot) {
 			var	undef, token, bbcode, content, html, needsBlockWrap,
 				blockWrapOpen, isInline, lastChild,
-				ret = [];
+				ret = '';
 
```

Comment:
Do we have to substantiate these as strings in the first place?  Can we just leave them undefined?

Reply:
Sadly yes, appending a string to an undefined variable causes `undefined` to first be converted to the string  `"undefined"` before appending the other string. E.g.:

```js
var ret;
ret += 'test'; // ret will now equal 'undefinedtest' instead of 'test' as expected
```

We could replace the first concatenation with assignment but given all the if statements it would probably end up harder to read in this case, easier to initialise it to an empty string.